clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -324,44 +382,60 @@ public void commitAsync(final Map<TopicIdPartition, Acknowledgements> acknowledg
             Node node = cluster.nodeById(nodeId);
             if (node != null) {
                 Map<TopicIdPartition, Acknowledgements> acknowledgementsMapForNode = new HashMap<>();
+
+                acknowledgeRequestStates.putIfAbsent(nodeId, new Pair<>(null, null));
+
                 for (TopicIdPartition tip : sessionHandler.sessionPartitions()) {
                     Acknowledgements acknowledgements = acknowledgementsMap.get(tip);
                     if (acknowledgements != null) {
                         acknowledgementsMapForNode.put(tip, acknowledgements);
 
                         metricsManager.recordAcknowledgementSent(acknowledgements.size());
-                        log.debug("Added async acknowledge request for partition {} to node {}", tip.topicPartition(), node.id());
+                        log.debug("Added async acknowledge request for partition {} to node {}", tip.topicPartition(),
+                                node.id());
                         resultCount.incrementAndGet();
+                        if (acknowledgeRequestStates.get(nodeId).getFirst() == null) {
```

Comment:
You should assign the result of `acknowledgeRequestStates.get(nodeId).getFirst()` to a variable so that you can avoid repeatedly evaluating this expression. I think you're doing it 3 times when once would be enough and also make the code clearer.